src/doom/m_random.c

```diff
@@ -63,3 +63,10 @@ void M_ClearRandom (void)
 {
     rndindex = prndindex = 0;
 }
+
+// inspired by the same routine in Eternity, thanks haleyjd
+int P_SubRandom (void)
+{
+    int r = P_Random();
```

Comment:
In case you aren't familiar with them, you'll want to read up on [sequence points](https://en.wikipedia.org/wiki/Sequence_point) for some context on why this change is necessary.